### PR TITLE
Refactoring: renaming and hot-fixes

### PR DESCRIPTION
**Pull request by malukenho**
- [x] Use of a more strict cs
- [x] Add phpunit config file
- [x] Fix phpunit assertions parameter order
- [x] Upgrade to php8
- [x] Rename `ApiStatus` to `ApiThrottling`
- [x] Use https on api calls
- [x] Add a marker interface for exceptions
- [x] Simplify array checkings

### CODE REVIEW

File: src/IpDetails.php [hunk 1]
```diff
@@ -36,31 +38,45 @@ private function __construct(
         string $threatLevel,
         string $isp
     ) {
-        $this->ipAddress = $ipAddress;
+        $this->ipAddress     = $ipAddress;
```
Comment (agustingomes) [inline src/IpDetails.php:41]: I would transform this into a Constructor property promotion

Comment (ScullWM) [reply src/IpDetails.php:41]: Yes, now that we support php8 it would be a nice improvement 👍

**Comment by ScullWM** [comment]
Awesome work @malukenho ! Thanks <3

File: src/IpDetails.php (hunk 1 above)
Comment (agustingomes) [reply src/IpDetails.php:41]: Solved in https://github.com/ScullWM/dbip-client/pull/2/files

**Pull request merged**